atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/PersistentLockService.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.persistentlock;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+
+/**
+ * Provides endpoints for acquiring and releasing the Deletion Lock. This is intended to be used by backups and sweep,
+ * to ensure that only one of these operations is deleting data at once.
+ */
+@Path("/persistent-lock")
```

Comment:
Just managed to catch up with @jboreiko about this.

The reason for taking in the LockEntry is, if you try to release the lock and it's not there any more, then _you can no longer guarantee that your e.g. backup was safe_, because someone cleared the lock out from under you. We want to know if that was the case, therefore we need the LockEntry that you took out on acquire.

What to do when starvation occurs (e.g. someone kills a sweep job while it's deleting) remains an open question.

As for extension, I think it makes sense to design for extension without making it too cumbersome to use at the moment. Therefore I propose:
* release remains unchanged
* acquire takes a query param instead of a path param

If we then decide to support a second type of lock, acquire will have a second query param (or a pathparam as in your example), which will default to `DeletionLock` for legacy support. Release will not need to change, because `LockEntry` is already future-proofed.